src/llm/model.py

```diff
@@ -0,0 +1,26 @@
+from langchain.chains.llm import LLMChain
+from langchain_ollama.llms import OllamaLLM
+from langchain_core.prompts import PromptTemplate
+from langchain_community.vectorstores import FAISS
+from src.llm.embedding_model import EMBEDDING_MODEL
+
+
+class LLM:
+    def __init__(self, llm__model_name: str, template: str) -> None:
+        self.llm_model = OllamaLLM(model=llm__model_name)
+        self.prompt = PromptTemplate(input_variables=["question", "context"], template=template)
+        self.llm_chain = LLMChain(prompt=self.prompt, llm=self.llm_model, verbose=True)
+        self.vector_db = None
+
+    def load_vector_db(self, vector_db_path: str, embedding_model: EMBEDDING_MODEL) -> None:

```

Comment:
This method together with create_vector_db can be the methods for VectorDB class and then there may be one class RAG which will connect VectorDB with LLM.